benches/core/matrix.rs

```diff
@@ -63,86 +66,108 @@ fn mat100_add_mat100(bench: &mut criterion::Criterion) {
+    bench.bench_function("mat6_mul_mat6", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat7_mul_mat7(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(7, 7);
     let b = DMatrix::<f64>::new_random(7, 7);
 
-    bench.bench_function("mat7_mul_mat7", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat7_mul_mat7", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat8_mul_mat8(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(8, 8);
     let b = DMatrix::<f64>::new_random(8, 8);
 
-    bench.bench_function("mat8_mul_mat8", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat8_mul_mat8", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat9_mul_mat9(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(9, 9);
     let b = DMatrix::<f64>::new_random(9, 9);
 
-    bench.bench_function("mat9_mul_mat9", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat9_mul_mat9", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat10_mul_mat10(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(10, 10);
     let b = DMatrix::<f64>::new_random(10, 10);
 
-    bench.bench_function("mat10_mul_mat10", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat10_mul_mat10", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat10_mul_mat10_static(bench: &mut criterion::Criterion) {
     let a = OMatrix::<f64, U10, U10>::new_random();
     let b = OMatrix::<f64, U10, U10>::new_random();
 
-    bench.bench_function("mat10_mul_mat10_static", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat10_mul_mat10_static", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat100_mul_mat100(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(100, 100);
     let b = DMatrix::<f64>::new_random(100, 100);
 
-    bench.bench_function("mat100_mul_mat100", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat100_mul_mat100", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn mat500_mul_mat500(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::from_element(500, 500, 5f64);
     let b = DMatrix::<f64>::from_element(500, 500, 6f64);
 
-    bench.bench_function("mat500_mul_mat500", move |bh| bh.iter(|| &a * &b));
+    bench.bench_function("mat500_mul_mat500", move |bh| {
+        bh.iter(|| black_box(&a) * black_box(&b))
+    });
 }
 
 fn iter(bench: &mut criterion::Criterion) {
     let a = DMatrix::<f64>::new_random(1000, 1000);
 
     bench.bench_function("iter", move |bh| {
         bh.iter(|| {
-            for value in a.iter() {
-                std::hint::black_box(value);
+            for value in black_box(&a).iter() {
```

Comment:
are you just being cautious here or is it really necessary to `black_box` the matrix reference before calling `iter` on it?

Reply:
I do not think that Rust compiler or LLVM are able to optimize anything significant out in this particular case with matrix that large. So, just a precaution plus code consistency.